### ISSUE TITLE
Style book: Close button and tabs visual order and DOM order mismatch

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#52730]
### Description

Go to the Site Editor > Styles > Style Book.
The editor switches to the so called 'focus mode'. 
Focus is moved to the X close button that is visually placed to the top right of the focus mode container.
Press the Tab key.
Observe that focus jumps to the left, to the first tab within the focus mode container.
Screenshot:

<img width="1150" alt="Screenshot 2023-07-18 at 14 35 04" src="https://github.com/WordPress/gutenberg/assets/1682452/7ef05002-6163-4ade-ab5c-4f374e6ae525">

The reversed order of the tab sequence is less than ideal for accessibility. Unexpected for keyboard users and problematic for screen magnification tools users. VIsual order and DOM order should always match when they affect meaning and operability / interaction. A mismatch is a WCAG violation.

### Step-by-step reproduction instructions

See above in the issue description.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
